kvmagent/kvmagent/plugins/vm_plugin.py

```
@@ -6949,6 +6963,35 @@ def make_mdev_device(mdevUuids):
+                    shmem_size = getattr(shmem, 'size', None)
+                if not mem_path:
+                    raise kvmagent.KvmError('dGPU shmem path is required but missing in StartVmCmd.addons[pciDevice.dgpu]')
+                normalized_mem_path = os.path.normpath(mem_path)
+                if not re.match(r'^/dev/shm/[a-zA-Z0-9_-]+$', normalized_mem_path):
+                    raise kvmagent.KvmError('invalid dGPU shmem path[%s], expected /dev/shm/*' % mem_path)
+                if normalized_mem_path != mem_path:
+                    raise kvmagent.KvmError('invalid dGPU shmem path[%s], path must be normalized under /dev/shm' % mem_path)
+                try:
+                    shmem_size = int(shmem_size)
+                except (TypeError, ValueError):
+                    raise kvmagent.KvmError('invalid dGPU shmem size[%s], must be an integer' % shmem_size)
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)